### DISCUSSION TITLE
Confusion about deploying a Next.js project with Node

### DISCUSSION THREAD

**Discussion by stackerito** [vercel/next.js#28066] [Help] [answered]
I remember that I had to use something like http-server or express in Node to serve pages.

But when deploying a Next.js project, there is no such thing, you just run the `npm run build` and then `npm run start` and your Node server is running and your app is served (from the docs):
https://nextjs.org/learn/basics/deploying-nextjs-app/other-hosting-options

So where does the "server" come into play here? I'm used to having some sort of server program to handle requests, such as Apache with PHP

**Comment by jamesmosier** [comment, marked as answer]
Next.js contains the Node server that serves your application. Therefore you do not need any extra web server in order to serve your Next.js application. 

Running `npm run start` starts up a Node server on a port and then it is up to your other hosting option to expose that port "publicly".

**Reply by buzztnt** [reply]
Hello @jamesmosier thanks for the clarification I myself am also stuck on this - do I upload the entire project folder or just the .next build folder?
